client/modules/IDE/components/Editor/index.jsx

```diff
@@ -81,36 +76,128 @@ window.HTMLHint = HTMLHint;
 
 const INDENTATION_AMOUNT = 2;
 
-class Editor extends React.Component {
-  constructor(props) {
-    super(props);
-    this.tidyCode = this.tidyCode.bind(this);
+function Editor(props) {
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+  // Defining states to usee
+  const files = useSelector((state) => state.files);
+  const ide = useSelector((state) => state.ide);
+  const file = useSelector(selectActiveFile);
+  const project = useSelector((state) => state.project);
+  const preferences = useSelector((state) => state.preferences);
+  const editorAccessibility = useSelector((state) => state.editorAccessibility);
+  const sidebarIsExpanded = useSelector((state) => state.ide.sidebarIsExpanded);
+  const consoleEvents = useSelector((state) => state.console);
+  const mergedState = {
+    ...preferences,
+    ...ide,
+    ...project,
+    ...editorAccessibility,
+    isExpanded: sidebarIsExpanded
+  };
+  // Binding Action Creators
+  const funcprop = bindActionCreators(
+    Object.assign(
+      {},
+      EditorAccessibilityActions,
+      FileActions,
+      ProjectActions,
+      IDEActions,
+      PreferencesActions,
+      UserActions,
+      ConsoleActions
+    ),
+    dispatch
+  );
```

Comment:
I wouldn't bother binding anything here.  I would call `dispatch(clearLintMessage())` etc. directly.

There could potentially be a slight benefit to binding if you were to do it inside a `useMemo` such that it only gets executed once.  In that case you would have a stable prop for the places where you pass an action creator down like `onClick={funcprop.expandSidebar}`.  But with the way that it is currently you'll get a new function on each render regardless, so you can write `onClick={() => dispatch(expandSidebar())}` and it's the same as what you have now.